server/apps/packages/package_service.py

```diff
@@ -255,6 +254,19 @@ def remove_refs_in_package(self, package, ref_id_to_remove, processed_packages=N
                 break
 
         updates = {config.LAST_UPDATED: utcnow(), 'groups': new_groups}
+
+        # if takes package then adjust the reference.
+        # safe to do this as take can only be in one takes package.
+        if package.get(PACKAGE_TYPE) == TAKES_PACKAGE:
+            new_sequence = package[SEQUENCE] - 1
+            updates[SEQUENCE] = new_sequence
+            last_take_group = [reference for reference in
+                               [new_group.get('refs') for new_group in new_groups if new_group['id'] == 'main'][0]
```

Comment:
This inner temporary list is kind of redundant, it can be converted to a generator (and simply taking the first item it yields from it), e.g.:

```python
next(new_group.get('refs') for new_group in new_groups if new_group['id'] == 'main')
```